### PR TITLE
[WIP] com.rstudio.RStudio

### PR DESCRIPTION
**Pull request by kevinsmia1939**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **https://github.com/rstudio/rstudio/issues/3079**
- [ ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by kevinsmia1939** [comment]
bot, build com.rstudio.RStudio

**Comment by kevinsmia1939** [comment]
bot, build

**Comment by kevinsmia1939** [comment]
bot, build com.rstudio.RStudio

**Comment by kevinsmia1939** [comment]
bot, build com.rstudio.RStudio

File: install-desktop-icon-dest.patch [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
++           DESTINATION /app/share/icons/hicolor/32x32/apps)
+    install(FILES ${RSTUDIO_ICONS_48}/rstudio.png
+-           DESTINATION /usr/share/icons/hicolor/48x48/apps)
++           DESTINATION /app/share/icons/hicolor/48x48/apps)
+    install(FILES ${RSTUDIO_ICONS_256}/rstudio.png
+-           DESTINATION /usr/share/icons/hicolor/256x256/apps)
++           DESTINATION /app/share/icons/hicolor/256x256/apps)
+ 
+    # .RData icon
+    install(FILES ${RSTUDIO_ICONS_16}/application-x-r-data.png
```
[the remark this answers is not in the document]
Comment (jurf) [reply install-desktop-icon-dest.patch:42]: Yes.

**Comment by kevinsmia1939** [comment]
The progress is halt for now, since RStudio pull in external plugins and require extra dependencies to build them which might be difficult

**Comment by jurf** [comment]
Shouldn’t having R-lang installed be enough?

**Comment by hfiguiere** [comment]
there are already a few packages on Falthub to include R.

**Comment by kevinsmia1939** [comment]
> Shouldn’t having R-lang installed be enough?

Many of the plugins in RStudio is downloaded as source and RStudio compile it, which need more dependencies. If the plugin download as binary it might be better for us.

**Pull request closed without merging**